R/antigen.garnish_run.R

```diff
@@ -171,6 +171,135 @@ ag_data_err <- function(){
 
 }
 
+#' Internal function to configure netMHC suite of tools
+#'
+#' @param dir Character vector. Path to `antigen.garnish` data directory.
+#'
+#' @export configure_netMHC_tools
+#' @md
+
+configure_netMHC_tools <- function(dir){
+
+  owd <- getwd()
+
+  on.exit({setwd(owd)})
+
+  # netMHC parent dir in ag data dir
+  npd <- file.path(dir, "netMHC")
+
+  if (!dir.exists(npd)) dir.create(npd)
+
+  # get path to scripts
+  setwd(npd)
+
+  tool_paths <- list.files()
+
+  f <- lapply(tool_paths, function(i){
+
+    # this pattern could one day break.
+    list.files(i, pattern = "netMHC(II)?(pan)?-?([0-9]\\.[0-9])?$",
+        full.names = TRUE)
+
+  }) %>% unlist
+
+  message("Checking netMHC scripts in antigen.garnish data directory.")
+  # sed scripts to correct paths
+  # install data from DTU
+  io <- lapply(f, function(i){
+
+    # rename to take off version, necessary because commands are built into table
+    # in get_pred_commands and check_pred_tools hasn't run at that point
+    io <- file.path(dirname(i),
+    basename(i) %>% stringr::str_replace("-.*$", ""))
+
+    # check if scripts were already edited
+    if (file.exists(file.path(dirname(i), "itwasedited.txt")))
```

Comment:
it would be safer to directly search for the correct patterns and change it they don't exist